core/src/main/java/nu/yona/server/messaging/service/FirebaseService.java

```diff
@@ -112,14 +115,14 @@ private long getMessageId(nu.yona.server.messaging.entities.Message message)
 		return Optional.ofNullable(lastMessageByRegistrationToken.remove(registrationToken));
 	}
 
-	private void sendMessage(String registrationToken, Message firebaseMessage)
+	private void sendMessage(String registrationToken, UUID deviceAnonymizedId, Message firebaseMessage)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This method has 25 lines, which is greater than the 20 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS138)

Reply:
IMHO, the method readability would decrease if we would further break down this method.

Reply:
We can extract the part marked with `// Store for testability` into a method `storeForTest(registrationToken, firebaseMessage)`

Reply:
OK, that helps the readability too.